http/src/request/channel/message/create_message.rs

```diff
@@ -39,10 +58,28 @@ impl<'a> CreateMessage<'a> {
         }
     }
 
-    pub fn content(mut self, content: impl Into<String>) -> Self {
-        self.fields.content.replace(content.into());
+    /// Set the content of the message.
+    ///
+    /// The maximum length is 2000 UTF-8 characters.
```

Comment:
The maximum length on discord is actually 2000 UTF-16 codepoints as they use python which calculates it in that way.

Though this may be a bit pedantic and I am not sure how to calculate the limit
in that case.

https://github.com/discord/discord-api-docs/issues/1315#issuecomment-577461770
https://stackoverflow.com/a/30775818

Reply:
After looking a bit more into it, It will still have the same amount of codepoints, but a different amount of code units. so the only thing left to resolve is if we should say utf-16 instead of utf-8.